### PR TITLE
Debug test refactor

### PR DESCRIPTION
**Pull request by davemay99**
Refactored tests to eliminate duplicated code, and added two writeBytes tests to validate writing a nil file and path escaping sandbox.  Tests that do not use the runTestCases function have been moved to the bottom of the file.

### CODE REVIEW

File: command/operator_debug_test.go [hunk 1]
```diff
@@ -275,136 +183,152 @@ func TestDebug_ClientToServer(t *testing.T) {
...
-	require.Empty(t, ui.ErrorWriter.String(), "errorwriter should be empty")
-	require.Contains(t, ui.OutputWriter.String(), "Starting debugger")
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+	runTestCases(t, cases)
 }
 
-func TestDebugFails(t *testing.T) {
+//////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
```
Comment (tgross) [inline command/operator_debug_test.go:287]: We don't typically have "banners" in our go code. I think this safe to leave out. Or if you want to keep it at least add some text under it like:

```go
// -------------------
// tests below don't use the test helper
```

File: command/operator_debug_test.go [hunk 2]
```diff
@@ -275,136 +183,152 @@ func TestDebug_ClientToServer(t *testing.T) {
...
+		pluginFile = filepath.Join(path, "nomad", "0000", pluginFile)
+		pluginFiles = append(pluginFiles, pluginFile)
+	}
+
+	testutil.WaitForFiles(t, pluginFiles)
 }
 
-func TestDebugCapturedFiles(t *testing.T) {
+func TestDebug_CapturedFiles(t *testing.T) {
 	// NOTE: pprof tracing/profiling cannot be run in parallel
```
Comment (tgross) [inline command/operator_debug_test.go:332]: Are there any tests in this file that this comment doesn't apply to? Maybe it should be a top-level comment?

File: command/operator_debug_test.go [hunk 3]
```diff
@@ -16,71 +16,37 @@ import (
...
+			code := cmd.Run(c.args)
+			out := ui.OutputWriter.String()
+			outerr := ui.ErrorWriter.String()
 
-	e = &external{addrVal: "127.0.0.1:8500", ssl: true}
-	require.Equal(t, "https://127.0.0.1:8500", e.addr("foo"))
+			// Verify case expectations
+			require.Equalf(t, code, c.expectedCode, "expected exit code %d, got: %d: %s", c.expectedCode, code, outerr)
+			for _, expectedOutput := range c.expectedOutputs {
+				require.Contains(t, out, expectedOutput, "expected output \"%s\", got \"%s\"", expectedOutput, out)
```
Comment (tgross) [inline command/operator_debug_test.go:45]: The `%q` formatter does this for you:

```suggestion
				require.Contains(t, out, expectedOutput, "expected output %q, got %q", expectedOutput, out)
```

File: command/operator_debug_test.go [hunk 4]
```diff
@@ -16,71 +16,37 @@ import (
...
+			}
+			require.Containsf(t, outerr, c.expectedError, "expected error \"%s\", got \"%s\"", c.expectedError, outerr)
```
Comment (tgross) [inline command/operator_debug_test.go:47]:
```suggestion
			require.Containsf(t, outerr, c.expectedError, "expected error %q, got %q", c.expectedError, outerr)
```

File: command/operator_debug_test.go [hunk 5]
```diff
@@ -16,71 +16,37 @@ import (
...
-	require.Empty(t, xs)
-
-	// address calculation honors CONSUL_HTTP_SSL
-	e := &external{addrVal: "http://127.0.0.1:8500", ssl: true}
-	require.Equal(t, "https://127.0.0.1:8500", e.addr("foo"))
+type testCases []testCase
 
-	e = &external{addrVal: "http://127.0.0.1:8500", ssl: false}
-	require.Equal(t, "http://127.0.0.1:8500", e.addr("foo"))
+func runTestCases(t *testing.T, cases testCases) {
```
Comment (tgross) [inline command/operator_debug_test.go:29]: :+1: on pulling this helper out.

**Review by tgross — APPROVED** [review]
Overall LGTM!

File: command/operator_debug_test.go (hunk 1 above)
Comment (davemay99) [reply command/operator_debug_test.go:287]: Removed

File: command/operator_debug_test.go (hunk 2 above)
Comment (davemay99) [reply command/operator_debug_test.go:332]: This is the only test which specifies an output directory, and looks for specific capture files to be present.

File: command/operator_debug_test.go (hunk 3 above)
Comment (davemay99) [reply command/operator_debug_test.go:45]: Hadn't seen that -- much cleaner, thanks!

**Pull request merged**